src/components/Todos/TodosList.tsx

```diff
@@ -50,14 +48,18 @@ const TodosList: FC = () => {
     } else if (activeTab === NavBarTabs.Completed) {
       setShowedTodos(completedTodos);
     }
-  }, [filteredTodos, activeTab]);
+  }, [todos, activeTab, searchValue]);
 
   useEffect(() => {
-    if (allTodos.length > prevTodosLength) {
+    if (todos.length > prevTodosLength) {
+      setActiveTab(NavBarTabs.All);
+      setPrevTodosLength(todos.length);
+    } else if (todos.length < prevTodosLength && !filteredTodos.length) {
       setActiveTab(NavBarTabs.All);
+      setPrevTodosLength(todos.length);
+      dispatch(searchTodos(""));
     }
```

Comment:
Can't you simplify this ?

Duplicates :
setActiveTab(NavBarTabs.All);
setPrevTodosLength(todos.length);